### PR TITLE
Publish/unpublish petition from edition page

### PR DESCRIPTION
**Pull request by lpoujade**
For issue #196, plus js to add an alert in case of failure (both on edition page and in dashboard)

### CODE REVIEW

File: pytition/petition/templates/petition/generic.js [hunk 1]
```diff
@@ -50,47 +50,34 @@ $(function () {
...
-            }, 1000);
-            //FIXME: show an alert message to the user about the failure
-        });
-    } else {
-        label.text("{% trans "Not published" %}");
-        custom_switch.removeClass("text-success");
-        custom_switch.addClass("text-danger");
-        $.ajax(petition_unpublish_url
-        ).done(function(){
+    var loader_id = 'loader_to_remove_' + Math.round(Math.random() * 100);
```
Comment (fallen) [inline pytition/petition/templates/petition/generic.js:61]: Why using random numbers here?
Just to make sure to generate unique id for the loader?
Why not adding a `data-petition-id="{{ petition.id }}"` to one of the parent div of the custom switch?
Like it's done for the publish and unpublish urls.

File: pytition/petition/templates/petition/generic.js [hunk 2]
```diff
@@ -50,47 +50,34 @@ $(function () {
...
-                custom_switch.addClass("text-success");
-                box.prop('disabled', false);
-            }, 1000);
-            //FIXME: show an alert message to the user about the failure
-        });
-    }
+            $(loader_id).remove();
+            var action = box.prop('checked') ? 'unpublish' : 'publish';
+            if (!box.siblings("div.alert").length)
+                $(`<div class="alert alert-danger" role="alert">Failed to ${action} the petition, try to refresh the page</div>`).insertAfter(label);
```
Comment (fallen) [inline pytition/petition/templates/petition/generic.js:79]: This works really great!
I wonder if we should remove this alert div if the user gets internet back and clicks again on the switch and this time it succeeds.
With your patch, when the switch publish/unpublish action succeeds again, the div alert stays.
Maybe I'm nitpicking here...

**Review by fallen — CHANGES_REQUESTED** [review]
This is a really good patch!
The UI change looks very good!
Congratz!
A few changes and then it can be merged.
Thanks a lot!

File: pytition/petition/templates/petition/generic.js (hunk 2 above)
Comment (lpoujade) [reply pytition/petition/templates/petition/generic.js:79]: you're right, if a user in on an unstable connection it can be useful

**Comment by fallen** [comment]
Hello @lpoujade , I hope you are well :)
If you find time to finish this PR it is still a valuable contribution in my opinion!

**Comment by lpoujade** [comment]
Hi, thank for the reminder ! I do this :)

**Comment by lpoujade** [comment]
Is it ok like this or did i have to add tests or something ?
Really sorry to not have done this before, that was not lot of work ...

**Review by fallen — APPROVED** [review]

**Comment by fallen** [comment]
Awesome! Thanks for the changes :)
No problem for the timing, everyone has its own schedule and time is a scarce resource! Thanks a lot!

**Pull request merged**